### ISSUE TITLE
Saving after changing anything on some forms causes VS to hang.

### ISSUE THREAD

**Issue by AhmadKelany** [dotnet/winforms#8952]
### Environment

Microsoft Visual Studio Community 2022
Version 17.5.3
VisualStudio.17.Release/17.5.3+33516.290
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Community

Visual C++ 2022   00482-90000-00000-AA004
Microsoft Visual C++ 2022

ASP.NET and Web Tools   17.5.318.41597
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.5.318.41597
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.5.318.41597
Azure Functions and Web Jobs Tools

C# Tools   4.5.0-6.23128.14+651d32d8017cfcc60fcbd67383601fa9a3deb145
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.4.3 (main@2a4517a)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.5.9 (11975e6)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.5.2.2316603+9f1b6856460af1e592d387ebef416eadddac453f
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.2.40119.0
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   17.0.20105.2003
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.5.0-6.23128.14+651d32d8017cfcc60fcbd67383601fa9a3deb145
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.5.0-beta.23053.5+794b7c259d9646a7eb685dad865aa27da7940a21
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.5.0.173 (d17-5@33e727c)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.5.3.46 (remotes/origin/d17-5@e4dd80b2bb)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.5.41 (ba80d05)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   13.2.0.0 (d17-5/797e2e1)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 6dd9def
    Java.Interop: xamarin/java.interop/main@149d70fe
    SQLite: xamarin/sqlite/3.40.0@fdc1e34
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@9f02d77

Xamarin.iOS and Xamarin.Mac SDK   16.2.0.5 (7738c90c9)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

### .NET version

<TargetFramework>net7.0-windows10.0.19041.0</TargetFramework>

### Did this work in a previous version of Visual Studio and/or previous .NET release?

Yes, the last one I am sure it worked with is 17.4. with same .Net release.

### Issue description

![Animation](https://user-images.githubusercontent.com/12924528/230602332-9980e00c-c630-41f2-b8ef-670847105211.gif)

In **some** forms, changing anything either by designer or by the properties window, then trying to save causes Visual Studio to hang.

### Steps to reproduce

The issue is not present in all forms, only some and I couldn't isolate the criteria based on which this is happening.

---------------------------------------------------

In new project: the issue doesn't exist.
My project was originally a .Net framework project that I ported to .Net 5 several years ago, and moved to .Net 7 when it came.
The issue appears in some old forms, new forms created do not have this issue.
However, copying some controls from a form that suffers from the issue to a new form even in a new project causes the problem to appear there also.

Here is the .Designer.cs code from a form that has the issue:

```
namespace UltimatePharmaForms;

partial class frmAdjustTax
{

    /// <summary>
    /// Clean up any resources being used.
    /// </summary>
    /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
    protected override void Dispose(bool disposing)
    {
        try
        {
            if (disposing && components != null)
            {
                components.Dispose();
            }
        }
        finally
        {
            base.Dispose(disposing);
        }
    }
    private System.ComponentModel.IContainer components = null;


    /// <summary>
    /// Required method for Designer support - do not modify
    /// the contents of this method with the code editor.
    /// </summary>
    private void InitializeComponent()
    {
        Label1 = new Label();
        Label2 = new Label();
        btnSave = new Button();
        btnSave.Click += btnSave_Click;
        txtFrom = new TextBox();
        txtTo = new TextBox();
        SuspendLayout();
        // 
        // Label1
        // 
        Label1.AutoSize = true;
        Label1.Location = new Point(461, 16);
        Label1.Margin = new Padding(4, 0, 4, 0);
        Label1.Name = "Label1";
        Label1.RightToLeft = RightToLeft.Yes;
        Label1.Size = new Size(199, 34);
        Label1.TabIndex = 0;
        Label1.Text = "تعديل ضريبة الأصناف من:";
        // 
        // Label2
        // 
        Label2.AutoSize = true;
        Label2.Location = new Point(115, 16);
        Label2.Margin = new Padding(4, 0, 4, 0);
        Label2.Name = "Label2";
        Label2.RightToLeft = RightToLeft.Yes;
        Label2.Size = new Size(42, 34);
        Label2.TabIndex = 1;
        Label2.Text = "إلى:";
        // 
        // Button1
        // 
        btnSave.Location = new Point(12, 63);
        btnSave.Name = "btnSave";
        btnSave.Size = new Size(608, 40);
        btnSave.TabIndex = 2;
        btnSave.Text = "تعديل";
        btnSave.UseVisualStyleBackColor = true;
        // 
        // txtFrom
        // 
        txtFrom.Location = new Point(305, 12);
        txtFrom.Name = "txtFrom";
        txtFrom.Size = new Size(100, 42);
        txtFrom.TabIndex = 3;
        // 
        // txtTo
        // 
        txtTo.Location = new Point(12, 12);
        txtTo.Name = "txtTo";
        txtTo.Size = new Size(100, 42);
        txtTo.TabIndex = 4;
        // 
        // frmAdjustTax
        // 
        AutoScaleMode = AutoScaleMode.None;
        ClientSize = new Size(630, 113);
        Controls.Add(txtTo);
        Controls.Add(txtFrom);
        Controls.Add(btnSave);
        Controls.Add(Label2);
        Controls.Add(Label1);
        Font = new Font("Noto Naskh Arabic", 12F, FontStyle.Regular, GraphicsUnit.Point);
        Name = "frmAdjustTax";
        Text = "تعديل الضريبة على الأصناف";
        Load += frmAdjustTax_Load;
        ResumeLayout(false);
        PerformLayout();
    }

    private Label Label1;
    private Label Label2;
    private Button btnSave;
    private TextBox txtFrom;
    private TextBox txtTo;
}
```

**Comment by elachlan** [comment]
@Olina-Zhang has your team run into this issue?

**Comment by RussKie** [comment]
@AhmadKelany is there anything in the Output window?

**Comment by AhmadKelany** [comment]
> @AhmadKelany is there anything in the Output window? https://aka.ms/winforms/designer/troubleshooting

The Visual Studio stops responding completely, I cannot get any information from the output window.

**Comment by AhmadKelany** [comment]
@Olina-Zhang Thanks for following up,
Here is a link to the repo:

https://dev.azure.com/ahmadkelany/_git/Ultimate%20Pharma%20Forms?path=/UltimatePharmaForms/Forms/AddEditForms

frmAdjustTax and some other forms have the same issue, I've made the repo public temporarily for you to see the files, If there is a way to share the repo with you without making it public please tell me.

Thanks very much.

**Comment by elachlan** [comment]
@AhmadKelany I think you should be able to add Olina as a collaborator. That way it avoids it being public.

**Comment by AhmadKelany** [comment]
> @AhmadKelany I think you should be able to add Olina as a collaborator. That way it avoids it being public.
@Olina-Zhang If it is possible, please provide an email to invite you to the repo, If this is the right way:
![image](https://user-images.githubusercontent.com/12924528/230889649-36272fc1-b4a3-4056-acfa-b914cc02cfe8.png)

Thanks again.

**Comment by RussKie** [comment]
> @AhmadKelany I think you should be able to add Olina as a collaborator. That way it avoids it being public.
> @Olina-Zhang Olina Zhang (Beyondsoft Corporation) Vendor If it is possible, please provide an email to invite you to the repo, If this is the right way:

I don't think that's the correct way. I suggest you open a VS Feedback ticket (unless you already have) where you can provide more information discreetly to Microsoft only.
Once you have opened the ticket, pleas post the link to it here for the team to track.

**Comment by merriemcgaw** [comment]
@Olina-Zhang can you make sure we have the above feedback ticket assigned to us (@KlausLoeffelmann probably) and we will dig in further to find out the issue.

**Comment by KlausLoeffelmann** [comment]
@AhmadKelany, I have not much experience with RTL languages, to be honest. My question is: How is your expected designer experience here? Do you need to set the right direction for your locale implicitly or is it derived from system settings?

**Comment by AhmadKelany** [comment]
@KlausLoeffelmann 
In the big picture of the designer, it doesn't matter much, I just put the controls where I want them, and I don't set the locale for the whole application.
RTL matters in some cases where I have a label with punctuation, e.g.: 
Name: 
In a label containging text like this in English, there is no problem, but:
الاسم:
In Arabic the colon appears in the wrong place in the label so I have to set the Label's RTL property to true.

Sorry about the long talk, I just try to clarify as much as I can.
In short:
- I don't set project-level or form-level locale, and even my windows OS locale is set to English, with Arabic only for Unicode characters.
- I set RTL only in some controls when I need them, and rely on controls positions and tab-order for RTL flow.

**Comment by KlausLoeffelmann** [comment]
> Sorry about the long talk, I just try to clarify as much as I can.

Don't be sorry, I appreciate a good explanation - thanks for that!

Just to be sure: Can you just as an attempt try to comment every single `[control].RightToLeft = RightToLeft.Yes` settings (inside of `InitializeComponent`) before you open the Form in question in design mode, and see if the Designer still hangs then?

Thanks!

**Comment by KlausLoeffelmann** [comment]
@AhmadKelany, thanks for trying that.
Can you please quickly recap, how what exact change you needed to make? Was it a RTL setting inside a Form or was it inside a UserControl, that you used inside a Form?

Thanks for your help here - we're really appreciate!

@merriemcgaw, @JeremyKuhne: This is something we need to discuss urgently.

**Comment by AhmadKelany** [comment]
@KlausLoeffelmann , Thanks for following up with me,
It was an RTL setting directly inside a Form, there were no UserControls involved in this scenario.
However:
- A UserControl with no child control with RTL property set, can be changed and saved via the designer normally.
- Trying to set RTL to yes on a control inside a UserControl via **the designer** causes VS to hang.
- Setting RTL to yes (or even to No) for a control inside a UserControl via **the .Designer.cs** file doesn't cause VS to hang directly, but any further attempts at changing anything in the UserControl via the designer causes VS to hang,
i.e.: with the RTL value set to either Yes or No, as long as the line exists in .Designer.cs file.
- When the mentioned UserControl with the line setting the RTL property is hosted inside a form, attempting changes to the forms **doesn't** cause VS to hang, as long as said changes doesn't involve RTL setting.

**Comment by KlausLoeffelmann** [comment]
This is really valuable information which will help us to get to the root cause of this!
We really appreciate your help here!

**Comment by KlausLoeffelmann** [comment]
@Olina-Zhang, based on the recent repro information, can you try to repro this, and if it repros, create an issue in the WinForms Designer?

Please link the issues against each other.

Thanks!

**Comment by Olina-Zhang** [comment]
@KlausLoeffelmann We still cannot reproduce it refer to the recent repro information "Trying to set RTL to yes on a control inside a User Control via the designer causes VS to hang." We can do some modify by designer and save all successfully without hang after setting controls with RTL: yes by designer setting (VS properties window) in User Control and Form designer.

**Comment by elachlan** [comment]
@AhmadKelany Is this issue still happening?

**Comment by AhmadKelany** [comment]
> @AhmadKelany Is this issue still happening?

I am using Visual Studio 2022 version 17.7.6 now and I confirm that the issue no longer happens. 👌
